test/widgets/main_test.dart

```diff
@@ -0,0 +1,179 @@
+      await SupabaseManager.reloadCurrentProfile();
+    }
+
+    setUp(() {
+      SupabaseManager.setCurrentProfile(null);
+      SupabaseManager.supabaseClient.auth.signOut();
+    });
+
+    testWidgets('It shows Welcome when not logged in', (WidgetTester tester) async {
+      await pumpMaterial(tester, const AuthApp());
+
+      expect(find.byType(WelcomePage), findsOneWidget);
+    });
+
+    testWidgets('It shows MainApp when logged in', (WidgetTester tester) async {
+      await setSession();
+
+      await pumpMaterial(tester, const AuthApp());
+
+      expect(find.byType(MainApp), findsOneWidget);
+    });
+
+    testWidgets('It shows MainApp after signing in', (WidgetTester tester) async {
+      await pumpMaterial(tester, const AuthApp());
+
+      expect(find.byType(WelcomePage), findsOneWidget);
+
+      whenRequest(
+        processor,
+        urlMatcher: equals('/auth/v1/token?grant_type=password'),
+        bodyMatcher: equals({'email': 'motismitfahrapp@gmail.com', 'password': '?Pass123word'}),
+        methodMatcher: equals('POST'),
+      ).thenReturnJson(responseHash);
+
+      // Sending AuthChangeEvent.signedIn
+      await SupabaseManager.supabaseClient.auth.signInWithPassword(
+        email: 'motismitfahrapp@gmail.com',
+        password: '?Pass123word',
+      );
+
+      await tester.pump();
+
+      expect(find.byType(MainApp), findsOneWidget);
+    });
+
+    testWidgets('It shows WelcomePage after signing out', (WidgetTester tester) async {
+      await setSession();
+
+      await pumpMaterial(tester, const AuthApp());
+
+      expect(find.byType(MainApp), findsOneWidget);
+
+      // Sending AuthChangeEvent.signedOut
+      await SupabaseManager.supabaseClient.auth.signOut();
+
+      await tester.pump();
+
+      expect(find.byType(WelcomePage), findsOneWidget);
+    });
+
+    testWidgets('It shows ResetPasswordPage after clicking deeplink', (WidgetTester tester) async {
+      await pumpMaterial(tester, const AuthApp());
+
+      whenRequest(
+        processor,
+        urlMatcher: equals('/auth/v1/user?'),
+        methodMatcher: equals('GET'),
+      ).thenReturnJson(userHash);
+
+      await SupabaseManager.supabaseClient.auth.getSessionFromUrl(
+        Uri(host: 'localhost', port: 3000, queryParameters: {
+          'access_token': 'access_token',
+          'expires_in': '3600',
+          'refresh_token': 'refresh_token',
+          'token_type': 'token_type',
+          'type': 'recovery'
+        }),
+      );
+
+      await tester.pump();
+
+      final Finder resetPasswordPage = find.byType(ResetPasswordPage);
+      expect(resetPasswordPage, findsOneWidget);
+      final ResetPasswordPage page = resetPasswordPage.evaluate().first.widget as ResetPasswordPage;
+      page.onPasswordReset();
+
+      await tester.pump();
+
+      expect(find.byType(MainApp), findsOneWidget);
+    });
+
+    // TODO: Write test case that checks the snackbar when email is invalid/expired
```

Comment:
Hey @Eddie-42, I just tried to test this case.

The thing is: I don't think the case is reachable? I've looked into the code and it definitely seems impossible. I looked it up and saw that you added it with this commit: https://github.com/MOTIS-Mitfahr-App/flutter-app/commit/f7abadf01bf05991a633ce2844226a580c679678

Do you remember this? Did you try this out back then?

Reply:
I have no strong recollection, but I'm pretty sure I never tested it out. I think this might be copied from a site in the supabase docs or something. But why is it impossible to reach?

Reply:
I can show you in person, it's kind of difficult to explain on Github

Reply:
I just tested it, I could not get this case. So I'm just going to remove it. (And I could not find the site in the docs where you could have copied this from).